src/ripple/rpc/handlers/AMMInfo.cpp

```diff
@@ -78,32 +78,54 @@ doAMMInfo(RPC::JsonContext& context)
 
     Issue issue1;
     Issue issue2;
+    uint256 ammID;
+    bool const isAssets =
+        params.isMember(jss::asset) && params.isMember(jss::asset2);
+    bool const isAMMAccount = params.isMember(jss::amm_account);
 
-    if (!params.isMember(jss::asset) || !params.isMember(jss::asset2))
+    if ((isAssets && isAMMAccount) || (!isAssets && !isAMMAccount))
     {
         RPC::inject_error(rpcINVALID_PARAMS, result);
         return result;
     }
 
-    if (auto const i = getIssue(params[jss::asset], context.j); !i)
-    {
-        RPC::inject_error(i.error(), result);
+    std::shared_ptr<ReadView const> ledger;
+    result = RPC::lookupLedger(ledger, context);
+    if (!ledger)
         return result;
+
+    if (isAssets)
+    {
+        if (auto const i = getIssue(params[jss::asset], context.j); !i)
+        {
+            RPC::inject_error(i.error(), result);
+            return result;
+        }
+        else
+            issue1 = *i;
+        if (auto const i = getIssue(params[jss::asset2], context.j); !i)
+        {
+            RPC::inject_error(i.error(), result);
+            return result;
+        }
+        else
+            issue2 = *i;
     }
     else
-        issue1 = *i;
-    if (auto const i = getIssue(params[jss::asset2], context.j); !i)
     {
-        RPC::inject_error(i.error(), result);
-        return result;
+        if (auto const id = getAccount(params[jss::amm_account], result); !id)
+        {
+            RPC::inject_error(rpcACT_MALFORMED, result);
+            return result;
+        }
+        else if (auto const sle = ledger->read(keylet::account(*id)); !sle)
+        {
+            RPC::inject_error(rpcACT_MALFORMED, result);
+            return result;
+        }
+        else
+            ammID = sle->getFieldH256(sfAMMID);
```

Comment:
I think this code is clearer without the cascading local variables declared in the if statements. That pattern is OK if it prevents the declared variables from polluting the parent scope. Here the parent scope can be small.